dali/base/dautils.cpp

```diff
@@ -3672,21 +3685,30 @@ void remapGroupsToDafilesrv(IPropertyTree *file, INamedGroupStore *resolver)
                 group.setown(createIGroup(cluster.queryProp("Group")));
             else
             {
-                assertex(resolver);
+                // JCSMORE only expected here if via foreign access (not entirely sure if this route is ever possible anymore)
+                assertex(foreign);
                 StringBuffer defaultDir;
                 GroupType groupType;
-                group.setown(resolver->lookup(planeName, defaultDir, groupType));
+                group.setown(queryNamedGroupStore().lookup(planeName, defaultDir, groupType));
             }
 
             std::vector<INode *> nodes;
             if (isContainerized())
             {
                 Linked<INode> dafileSrvNodeCopy;
                 {
-                    // in case config hook above changes dafileSrvNode
+                    // in case config hook above changes tlsDirectIONode/nonTlsDirectIONode
                     CriticalBlock b(dafileSrvNodeCS);
-                    dafileSrvNodeCopy.set(dafileSrvNode);
+                    dafileSrvNodeCopy.set(secure ? tlsDirectIONode : nonTlsDirectIONode);
                 }
+                if (!dafileSrvNodeCopy)
+                {
+                    const char *typeText = secure ? "secure" : "non-secure";
+                    throw makeStringExceptionV(0, "%s DFS service request made, but no %s directio service available", typeText, typeText);
+                }
+                StringBuffer epStr;
+                DBGLOG("Remap secure=%s to: %s", boolToStr(secure), dafileSrvNodeCopy->endpoint().getEndpointHostText(epStr).str());
```

Comment:
@jakesmith will this generate excessive tracing?

Reply:
removed/squashed